servicedirectory/endpoints_test.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package servicedirectory
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+)
+
+func TestEndpoint(t *testing.T) {
+	tc := testutil.SystemTest(t)
+	if err := createNamespace(ioutil.Discard, tc.ProjectID); err != nil {
+		fmt.Printf("Failed to create namespace in test setup. Skipping: %v", err)
+		os.Exit(0)
+	}
+	if err := createService(ioutil.Discard, tc.ProjectID); err != nil {
```

Comment:
Can we combine these two test files into one big test (`endpoints_test.go` and `services_test.go`)? They both create a namespace, create a service, do a couple other things, then delete the namespace & service.

Also, I'm still thinking these should be failures. If the snippet fails to create a namespace/service, we want to fix the snippet. Otherwise, these functions aren't really tested.

If we did want to skip (I don't think we should), we should use `t.Skip`, not `fmt.Printf` & `os.Exit`.

Reply:
The reason why I had separated them to start was to keep namespace creation as part of setup as it is necessary to create services and endpoints.  I can make namespace creation a test as part of the larger test file but will have to skip the rest of the tests if it fails